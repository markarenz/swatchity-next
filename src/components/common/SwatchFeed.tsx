import React, { useState, useEffect } from 'react';
import Layout from '../pageComponents/Layout';
import { useIntl } from 'react-intl';
import { UserMeta } from '@prisma/client';
import { SwatchExt, Color, PageMeta, SubNavItem, UserProfile } from '@/types';
import { swatchesPerPage } from '@/constants';
import SwatchPost from './SwatchPost';
import SwatchSkeleton from './SwatchSkeleton';
import ColorPicker from './ColorPicker';
import { FormattedMessage } from 'react-intl';
import { getSwatches, createSwatch } from '@/utils/apiFunctions';
import { getRandomColor } from '@/utils/colorFunctions';
import ButtonIcon from '@/components/common/ButtonIcon';
import IconNew from '@/components/icons/IconNew';
import { useRouter } from 'next/router';
import IconRefresh from '@/components/icons/IconRefresh';
import SearchColor from '@/components/common/SearchColor';
import UserProfileBlock from '@/components/common/UserProfileBlock';

type Props = {
  subNavData: SubNavItem[];
  userMeta: UserMeta | null;
  initialSwatches: SwatchExt[];
  initialLikes: string[];
  mode: string;
  str: string;
  titleKey: string;
  introKey: string;
  userProfile?: UserProfile;
};
const SwatchFeed: React.FC<Props> = ({
  subNavData,
  initialSwatches,
  initialLikes,
  userMeta,
  mode,
  str,
  titleKey,
  introKey,
  userProfile,
}) => {
  const [isLoading, setIsLoading] = useState(false);
  const [swatches, setSwatches] = useState<SwatchExt[]>(initialSwatches);
  const [userLikes, setUserLikes] = useState<string[]>(initialLikes);
  const [isNewSwatchOpen, setIsNewSwatchOpen] = useState(false);
  const [canLoadMore, setCanLoadMore] = useState(true);
  const { formatMessage } = useIntl();
  const router = useRouter();
  const isLoggedIn = !!userMeta?.name;
  const isSearch = router.route === '/search/[rgb]';
  const newSwatchOpen = () => {
    setIsNewSwatchOpen(true);
  };
  const closeNewSwatch = () => {
    setIsNewSwatchOpen(false);
  };
  const pageTitle =
    mode === 'profile'
      ? formatMessage({ id: titleKey }, { name: `${userProfile?.name}` })
      : formatMessage({ id: titleKey });
  const pageIntro =
    mode === 'profile'
      ? formatMessage({ id: introKey }, { name: `${userProfile?.name}` })
      : formatMessage({ id: introKey });
  const pageMeta: PageMeta = {
    title: pageTitle,
    metadesc: pageIntro,
  };
  const createNewSwatch = async (color: Color) => {
    if (userMeta && userMeta.email) {
      const newSwatch = await createSwatch(userMeta.email, color.r, color.g, color.b);
      if (newSwatch) {
        setSwatches([newSwatch, ...swatches]);
      }
      closeNewSwatch();
      // if we are not on the home page, redirect to /
      if (router.pathname !== '/') {
        router.push('/');
      }
    }
  };
  const refreshSwatches = () => {
    setCanLoadMore(true);
    setSwatches([]);
    loadSwatches(0);
  };
  const loadMoreSwatches = () => {
    loadSwatches(swatches.length);
  };
  const loadSwatches = async (sk: number) => {
    if (userMeta?.id) {
      setIsLoading(true);
      const newSwatchData = await getSwatches(`${userMeta?.id}`, mode, str, sk);
      const { likes: newLikes, swatches: newSwatches } = newSwatchData;
      if (sk > 0) {
        setSwatches([...swatches, ...newSwatches]);
        setUserLikes([...userLikes, newLikes]);
      } else {
        setSwatches(newSwatches);
        setUserLikes(newLikes);
      }
      if (newSwatches.length === 0) {
        setCanLoadMore(false);
      }
      setIsLoading(false);
    }
  };
  // For Search pages, we need to refresh data on page load for multiple searches in a row
  let rgb = '';
  let searchColor = { r: 0, g: 0, b: 0 };
  if (isSearch) {
    rgb = `${router?.query?.rgb}`;
    const [tmpR, tmpG, tmpB] = rgb.split('-');
    searchColor = { r: parseInt(tmpR, 10), g: parseInt(tmpG, 10), b: parseInt(tmpB, 10) };
  }
  useEffect(() => {
    if (!!userMeta?.id) {
      refreshSwatches();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [rgb]);
  const handleSearchColorChange = (color: Color) => {
    const path = `/search/${color.r}-${color.g}-${color.b}`;
    router.push(path);
  };
  const headerButtons = (
    <React.Fragment>
      <ButtonIcon onClick={refreshSwatches} label={formatMessage({ id: 'feed__refresh' })}>
        <div key="btn-refresh" className="hover-zoom h-3 w-3 py-0-5 px-0-5">
          <IconRefresh color="gray-6" colorDark="gray-2" />
        </div>
      </ButtonIcon>
      {isLoggedIn && (
        <ButtonIcon
          key="btn-new"
          onClick={newSwatchOpen}
          label={formatMessage({ id: 'feed__new_swatch' })}
          testID="feed-btn-new"
        >
          <div className="hover-zoom h-3 w-3 py-0-5 px-0-5">
            <IconNew color="gray-6" colorDark="gray-2" />
          </div>
        </ButtonIcon>
      )}
    </React.Fragment>
  );
  const warningNotLoggedIn = !isLoggedIn && mode === 'liked';
  return (
    <Layout pageMeta={pageMeta} subNavData={subNavData} headerButtons={headerButtons}>
      <div className="contained pt-1 pb-4" id="swatch-feed">
        <div className="flex items-center">
          {mode === 'profile' && !!userProfile ? (
            <UserProfileBlock userProfile={userProfile} mode={mode} />
          ) : (
            <div className="flex-grow">
              <h1 className="text-3">{pageTitle}</h1>
              <div className="mb-1 italic">
                <p>{pageIntro}</p>
              </div>
            </div>
          )}
          {isSearch && (
            <div className="pl-2">
              <SearchColor color={searchColor} onChange={handleSearchColorChange} />
            </div>
          )}
        </div>
        <div id="feed" className="pt-1 pb-2">
          <div id="swatchPosts">
            {swatches.map((s) => (
              <SwatchPost
                userID={userMeta?.id}
                swatch={s}
                key={s.id}
                setUserLikes={setUserLikes}
                setSwatches={setSwatches}
                isLiked={userLikes.includes(s.id)}
              />
            ))}
          </div>

          {isLoading &&
            [...Array(swatchesPerPage)].map((_e, idx) => (
              <SwatchSkeleton key={`skeleton-swatch-${idx}`} isLoggedIn={isLoggedIn} />
            ))}
          {!isLoading && swatches.length === 0 && mode === 'search' && (
            <h2>
              <FormattedMessage id="search__no_results" />
            </h2>
          )}
          {warningNotLoggedIn && (
            <h2>
              <FormattedMessage id="feed__liked__not_logged_in" />
            </h2>
          )}
          {canLoadMore && !warningNotLoggedIn && (
            <div className="text-center">
              <button onClick={loadMoreSwatches} className="btn" data-testid="feed-load-more">
                <FormattedMessage id="feed__load_more" />
              </button>
            </div>
          )}
        </div>
        <ColorPicker
          color={getRandomColor()}
          isOpen={isNewSwatchOpen}
          closeColorPicker={closeNewSwatch}
          onChange={createNewSwatch}
        />
      </div>
    </Layout>
  );
};

export default SwatchFeed;
