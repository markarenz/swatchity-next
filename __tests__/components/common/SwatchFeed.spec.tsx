import SwatchFeed from '@/components/common/SwatchFeed';
import { render, screen, fireEvent, act, waitFor } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { useRouter } from 'next/router';
import messages from '@/locale/en-US.json';
import mockSwatch from '../../__fixtures__/mockSwatch';
import mockUserData from '../../__fixtures__/mockUserMeta';
import { subNavLinksFeed } from '@/constants';

const mockSwatchExt = {
  ...mockSwatch,
  user: {
    ...mockUserData,
  },
};

jest.mock('@/context/UserContext', () => ({
  useUserContext: jest.fn(() => ({
    userMeta: mockUserData,
  })),
}));

jest.mock('@/context/UserContext', () => ({
  useUserContext: jest
    .fn()
    .mockImplementation(() => ({ userMeta: mockUserData }))
    .mockImplementationOnce(() => ({ userMeta: null })),
}));

jest.mock('next/router', () => ({
  useRouter: jest.fn(() => ({
    route: '/',
    pathname: '/',
    push: jest.fn(),
  })),
}));

jest.mock('next-auth/react', () => ({
  useSession: jest
    .fn()
    .mockReturnValueOnce({
      data: undefined,
    })
    .mockReturnValue({
      data: {
        email: 'email@test.com',
        name: 'Test Name',
      },
    }),
}));

jest.mock('@/utils/apiFunctions', () => ({
  ...jest.requireActual('@/utils/apiFunctions'),
  getSwatches: jest
    .fn()
    .mockImplementation(() => ({
      likes: ['5678'],
      swatches: [
        {
          ...mockSwatch,
          id: '5678',
        },
      ],
    }))
    .mockImplementationOnce(() => ({
      likes: [],
      swatches: [],
    })),

  createSwatch: jest.fn(() => ({
    ...mockSwatchExt,
  })),
}));

const mockProps = {
  subNavData: subNavLinksFeed,
  initialSwatches: [mockSwatchExt],
  initialLikes: ['abcd1234'],
  userMeta: mockUserData,
  mode: 'feed',
  str: '',
  titleKey: 'feed__home__title',
  introKey: 'feed__home__intro',
};
beforeEach(() => {
  jest.resetModules();
});
afterEach(() => {
  jest.restoreAllMocks();
});

describe('SwatchFeed', () => {
  it('renders component', async () => {
    render(
      <IntlProvider messages={messages} locale="en" defaultLocale="en">
        <SwatchFeed {...mockProps} />
      </IntlProvider>,
    );
  });

  it('handles open & close color picker', async () => {
    act(() => {
      render(
        <IntlProvider messages={messages} locale="en" defaultLocale="en">
          <SwatchFeed {...mockProps} />
        </IntlProvider>,
      );
    });
    const btnNew = await screen.findByTestId('feed-btn-new');
    act(() => {
      fireEvent(
        btnNew,
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );
    });
    const btnClose = document.querySelector('#swatch-feed .colorPicker .pickerClose');
    if (btnClose) {
      fireEvent(
        btnClose,
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );
    }
    expect(screen).toMatchSnapshot();
  });

  it('handles create new color', async () => {
    act(() => {
      render(
        <IntlProvider messages={messages} locale="en" defaultLocale="en">
          <SwatchFeed {...mockProps} />
        </IntlProvider>,
      );
    });
    const btnNew = await screen.findByTestId('feed-btn-new');
    act(() => {
      fireEvent(
        btnNew,
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );
    });
    const btnOK = document.querySelector('#swatch-feed .colorPicker .pickerOK');
    if (btnOK) {
      fireEvent(
        btnOK,
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );
    }
    expect(screen).toMatchSnapshot();
  });

  it('handles create new color - other feed', async () => {
    // @ts-ignore
    useRouter.mockImplementation(() => ({
      route: '/feed/liked',
      pathname: '/feed/liked',
      push: jest.fn(),
    }));
    act(() => {
      render(
        <IntlProvider messages={messages} locale="en" defaultLocale="en">
          <SwatchFeed {...mockProps} />
        </IntlProvider>,
      );
    });
    const btnNew = await screen.findByTestId('feed-btn-new');
    act(() => {
      fireEvent(
        btnNew,
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );
    });
    const btnOK = document.querySelector('#swatch-feed .colorPicker .pickerOK');
    if (btnOK) {
      fireEvent(
        btnOK,
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );
    }
    expect(screen).toMatchSnapshot();
  });

  it('handles load more button click - new swatches', async () => {
    act(() => {
      render(
        <IntlProvider messages={messages} locale="en" defaultLocale="en">
          <SwatchFeed {...mockProps} />
        </IntlProvider>,
      );
    });
    await waitFor(() => {
      const swatchPosts = document.querySelector('#swatchPosts')?.childNodes;
      expect(swatchPosts && swatchPosts.length > 0).toBe(true);
    });
    const btnLoad = await screen.findByTestId('feed-load-more');
    act(() => {
      fireEvent(
        btnLoad,
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );
    });
    expect(screen).toMatchSnapshot();
  });
  it('handles load more button click - no more swatches', async () => {
    act(() => {
      render(
        <IntlProvider messages={messages} locale="en" defaultLocale="en">
          <SwatchFeed {...mockProps} />
        </IntlProvider>,
      );
    });
    const btnLoad = await screen.findByTestId('feed-load-more');
    act(() => {
      fireEvent(
        btnLoad,
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );
    });
    expect(screen).toMatchSnapshot();
  });
  it('handles search mode - change color', async () => {
    // @ts-ignore
    useRouter.mockImplementation(() => ({
      route: '/search/[rgb]',
      pathname: '/searh/10-110-210',
      push: jest.fn(),
    }));
    act(() => {
      render(
        <IntlProvider messages={messages} locale="en" defaultLocale="en">
          <SwatchFeed {...mockProps} mode="search" str="10-110-210" />
        </IntlProvider>,
      );
    });
    const btnColor = await screen.findByTestId('btn-search-color');
    act(() => {
      fireEvent(
        btnColor,
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );
    });
    const btnOK = document.querySelector('#swatch-feed .colorPicker .pickerOK');
    if (btnOK) {
      fireEvent(
        btnOK,
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );
    }
    expect(screen).toMatchSnapshot();
  });

  it('displays message in search mode - no results', async () => {
    act(() => {
      render(
        <IntlProvider messages={messages} locale="en" defaultLocale="en">
          <SwatchFeed {...mockProps} mode="search" str="" initialLikes={[]} initialSwatches={[]} />
        </IntlProvider>,
      );
    });
    expect(screen).toMatchSnapshot();
  });

  it('displays a warning when not logged in in liked mode', () => {
    act(() => {
      render(
        <IntlProvider messages={messages} locale="en" defaultLocale="en">
          <SwatchFeed {...mockProps} mode="liked" userMeta={null} />
        </IntlProvider>,
      );
    });
    expect(screen).toMatchSnapshot();
  });

  it('displays profile feed', async () => {
    act(() => {
      render(
        <IntlProvider messages={messages} locale="en" defaultLocale="en">
          <SwatchFeed {...mockProps} mode="profile" str="" userProfile={mockUserData} />
        </IntlProvider>,
      );
    });
    expect(screen).toMatchSnapshot();
  });
});
