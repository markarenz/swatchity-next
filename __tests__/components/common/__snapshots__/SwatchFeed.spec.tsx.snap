// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwatchFeed displays a warning when not logged in in liked mode 1`] = `
{
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`SwatchFeed displays message in search mode - no results 1`] = `
{
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`SwatchFeed displays profile feed 1`] = `
{
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`SwatchFeed handles create new color - other feed 1`] = `
{
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`SwatchFeed handles create new color 1`] = `
{
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`SwatchFeed handles load more button click - new swatches 1`] = `
{
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`SwatchFeed handles load more button click - no more swatches 1`] = `
{
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`SwatchFeed handles open & close color picker 1`] = `
{
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`SwatchFeed handles search mode - change color 1`] = `
{
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;
